glassfish 5.1.0

joaomlneto opened the pull request:
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

The test is fairly minimal, but I have no idea how to do them :-)
I could try to do something similar to the [wildfly-as formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/wildfly-as.rb)

-----

Glassfish has been [transfered to the Eclipse Foundation](https://www.eclipse.org/community/eclipse_newsletter/2019/february/GlassFish_5.1.php) from version 5.1

Currently there is no proper *homepage*. Candidates:
- https://github.com/eclipse-ee4j/glassfish
- https://projects.eclipse.org/projects/ee4j.glassfish

SMillerDev commented:
This test isn't doing much so please try your suggestion of copying the other test.

Just use the github page as homepage.

SMillerDev commented on Formula/glassfish.rb:
```diff
@@ -19,6 +19,12 @@ def install
   def caveats; <<~EOS
     You may want to add the following to your .bash_profile:
       export GLASSFISH_HOME=#{opt_libexec}
+      export PATH=${PATH}:${GLASSFISH_HOME}/glassfish/bin
```
We  don't suggest path changes in homebrew.

joaomlneto commented:
@SMillerDev I have addressed all your comments.

One potential issue with the testing is that Glassfish requires Java 8 to start (exactly version 8). Should I change the `depends_on` attribute?

fxcoudert commented:
```
==> brew test glassfish --verbose
==> FAILED
Testing glassfish
/usr/bin/sandbox-exec -f /private/tmp/homebrew20191226-75747-1mrwsex.sb ruby -W0 -I $LOAD_PATH -- /usr/local/Homebrew/Library/Homebrew/test.rb /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/glassfish.rb --verbose
==> /usr/local/opt/glassfish/libexec/glassfish/bin/asadmin version | grep 5.1.0
Picked up _JAVA_OPTIONS: -Duser.home=/Users/brew/Library/Caches/Homebrew/java_cache
Version = GlassFish Server Open Source Edition  5.1.0  (build default-private)
Picked up _JAVA_OPTIONS: -Duser.home=/Users/brew/Library/Caches/Homebrew/java_cache
Exception in thread "main" java.lang.NullPointerException
	at com.sun.enterprise.module.common_impl.AbstractModulesRegistryImpl.initializeServiceLocator(AbstractModulesRegistryImpl.java:128)
	at com.sun.enterprise.module.common_impl.AbstractModulesRegistryImpl.newServiceLocator(AbstractModulesRegistryImpl.java:120)
	at com.sun.enterprise.module.common_impl.AbstractModulesRegistryImpl.createServiceLocator(AbstractModulesRegistryImpl.java:194)
	at com.sun.enterprise.module.common_impl.AbstractModulesRegistryImpl.createServiceLocator(AbstractModulesRegistryImpl.java:200)
	at com.sun.enterprise.module.single.StaticModulesRegistry.createServiceLocator(StaticModulesRegistry.java:64)
	at com.sun.enterprise.admin.cli.CLIContainer.getServiceLocator(CLIContainer.java:193)
	at com.sun.enterprise.admin.cli.CLIContainer.getLocalCommand(CLIContainer.java:231)
	at com.sun.enterprise.admin.cli.CLICommand.getCommand(CLICommand.java:207)
	at com.sun.enterprise.admin.cli.AdminMain.executeCommand(AdminMain.java:347)
	at com.sun.enterprise.admin.cli.AdminMain.doMain(AdminMain.java:282)
	at org.glassfish.admin.cli.AsadminMain.main(AsadminMain.java:33)
```

joaomlneto commented:
@fxcoudert: That is most likely due to it being executed with a incompatible Java version[[1]](https://stackoverflow.com/questions/48546700/nullpointer-when-i-startup-glassfish-server) -- it works exclusively with Java 8.

I'm unsure how to encode this in the Formula. Any ideas?

This behavior was happening with the previous version of Glassfish.

joaomlneto commented:
I am now telling GlassFish to use JDK8 during the tests.

However, I'm doing this `/usr/libexec/java_home`, which seems to be MacOS-only (and not even sure if it is the *best* way of doing it).

Is there any HomeBrew recommendation on getting `JAVA_HOME` for a specific JDK?

fxcoudert commented:
We have `Language::Java.java_home_cmd("1.8")`, see how it's used in other formulas

zbeekman commented:
@joaomlneto any chance you could squash these commits and force push?

joaomlneto commented:
@zbeekman done :-)

---

###### many thanks for the HomeBrew project!
###### and thanks for putting up with my questions here!

chenrui333 commented:
Thanks @joaomlneto for your contribution to Homebrew! :tada: 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!
